logger.go

```diff
@@ -96,3 +112,72 @@ func (l *MatchingLogger) Write(p []byte) (int, error) {
 
 	return len(p), nil
 }
+
+// parsedLog is a lightweight log structure we parse from subsystem logs.
+// Another approach for capturing logs from subsystems is to pass around
+// LogEntry or opentelemetry structs.
```

Comment:
Ugh, this feels so... "wrong" to have to do it this way. Sorry, not saying you did it wrong, just, there should be a cleaner way in our logging system to handle this. A zap-style log line parser just doesn't feel like it should be living directly in the agent code. Also shouldn't be having to manually write to netappender either.

Do you think some of this would be better in the logging package itself? (Along with adding a Write(zapcore.Entry) function?)

Reply:
yeah agree re kludge

it's not a completely new problem; would imagine stackdriver, splunk, fluentd et al have to deal with this at some point as aggregators. wonder if there are are standard tools for this task

no preference about where it lives. if you want to move it can that be a fast-follow rather than a blocker?

Reply:
I feel it needs more thought, but its more important to get this out... so yeah, doesn't have to be a blocker on this PR.